manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorService.java

```diff
@@ -15,19 +16,21 @@ public interface ProcessorService {
 
     Processor createProcessor(String bridgeId, String customerId, String owner, ProcessorRequest processorRequest);
 
+    Processor updateProcessor(String bridgeId, String processorId, String customerId, ProcessorRequest processorRequest);
+
+    Optional<Processor> getErrorHandler(String bridgeId, String customerId);
+
     Processor createErrorHandlerProcessor(String bridgeId, String customerId, String owner, ProcessorRequest processorRequest);
 
-    Processor updateProcessor(String bridgeId, String processorId, String customerId, ProcessorRequest processorRequest);
+    Processor updateErrorHandlerProcessor(String bridgeId, String processorId, String customerId, ProcessorRequest processorRequest);
 
     List<Processor> findByShardIdWithReadyDependencies(String shardId);
 
     Processor updateProcessorStatus(ProcessorDTO processorDTO);
 
     Long getProcessorsCount(String bridgeId, String customerId);
 
-    ListResult<Processor> getUserVisibleProcessors(String bridgeId, String customerId, QueryProcessorResourceInfo queryInfo);
-
-    ListResult<Processor> getHiddenProcessors(String bridgeId, String customerId);
+    ListResult<Processor> getProcessors(String bridgeId, String customerId, QueryProcessorResourceInfo queryInfo);
```

Comment:
I like the `UserVisible`/`Hidden` naming to make clear that there are hidden processors, now this is not visible anymore but "implicit"

Reply:
@danielezonca We now only have `ListResult<Processor> getProcessors(..)` (all `SINK` and `SOURCE`) and `Optional<Processor> getErrorHandler(..)`. Everywhere we were using `getHiddenProcessors(..)` we examining the list for the first `ERROR_HANDLER`. We only support defining one Error Handler at the moment so I think this change is more aligned.